contracts/libraries/PathKey.sol

```diff
@@ -0,0 +1,14 @@
+//SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.20;
+
+import {Currency} from "@uniswap/v4-core/contracts/types/Currency.sol";
+import {IHooks} from "@uniswap/v4-core/contracts/interfaces/IHooks.sol";
+
+struct PathKey {
+    Currency intermediateCurrency;
```

Comment:
This concept of intermediateCurrency is a bit confusing since you have to know the prevToken which would require special casing when this is the first hop in a multi hop route and the final one. I wonder if it would be easier to make this `PathKey` more similar to the `PoolKey` type with the addition of `hookData`

Reply:
And as long as `zeroForOne` is right and the pools are in the right order it should work out?

Reply:
True although first/last hop from token wouldn't be in the path so it's a bit cleaner. I kinda lean toward leaving this as is just because we already got it working, unless there are other reasons? Curious what people think

Reply:
yea fair enough